tests/processor/test_main.py

```diff
@@ -0,0 +1,442 @@
+    assert 'vim-0:2-3.el7' == artifact.filename
+
+    # "re-creating" should just return existing node
+    artifact2 = analyzer.get_or_create_rpm_artifact_from_hash(VIM_2_3)
+    assert artifact.id == artifact2.id
+
+
+def test_get_or_create_archive_artifact():
+    """Test the basic function of the get_or_create_archive_artifact function."""
+    analyzer = main_analyzer.MainAnalyzer()
+    arch = 'x86_64'
+    artifact = analyzer.get_or_create_archive_artifact(
+        archive_id=IMAGE1['id'],
+        filename=IMAGE1['filename'],
+        arch=arch,
+        checksum=IMAGE1['checksum'])
+
+    assert IMAGE1['filename'] == artifact.filename
+    assert IMAGE1['checksum'] == artifact.checksum
+    assert 'archive-' + IMAGE1['id'] == artifact.archive_id
+    assert arch == artifact.architecture
+    artifact.id  # exists, hence is saved
+
+    # "re-creating" should just return existing node
+    artifact2 = analyzer.get_or_create_archive_artifact(
+        archive_id=IMAGE1['id'],
+        filename=IMAGE1['filename'],
+        arch=arch,
+        checksum=IMAGE1['checksum'])
+    assert artifact.id == artifact2.id
+
+
+def test_get_or_create_source_location():
+    """Test the basic function of the get_or_create_source_location function."""
+    analyzer = main_analyzer.MainAnalyzer()
+    url = 'www.whatever.com'
+    canonical_version = 'pi'
+    sl = analyzer.get_or_create_source_location(
+        url=url,
+        canonical_version=canonical_version)
+
+    assert sl.url == url
+    assert sl.canonical_version == sl.canonical_version
+    sl.id  # exists, hence is saved
+
+    # "re-creating" should just return existing node
+    sl2 = analyzer.get_or_create_source_location(
+        url=url,
+        canonical_version=canonical_version)
+    assert sl.id == sl2.id
+
+
+def test_get_or_create_component():
+    """Test the basic function of the get_or_create_component function."""
+    analyzer = main_analyzer.MainAnalyzer()
+    namespace = 'Pizza Hut'
+    name = 'Pepperoni'
+    type = 'pizza'
+    component = analyzer.get_or_create_component(
+        canonical_namespace=namespace,
+        canonical_name=name,
+        canonical_type=type)
+
+    assert component.canonical_namespace == namespace
+    assert component.canonical_name == name
+    assert component.canonical_type == type
+    component.id  # exists, hence is saved
+
+    # "re-creating" should just return existing node
+    component2 = analyzer.get_or_create_component(
+        canonical_namespace=namespace,
+        canonical_name=name,
+        canonical_type=type)
+    assert component.id == component2.id
+
+
+def good_run(self):
+    """Mock a simple run and succeed."""
+    self.get_or_create_build('1234', '1')
+
+
+@mock.patch('assayist.processor.main_analyzer.MainAnalyzer.run', new=good_run)
+def test_main_good():
+    """Ensure that the main function normally runs and commites successfully."""
```

Comment:
Typo in docstring

Reply:
Heh, I must need more coffee but I'm not seeing it. What word is misspelled?

Reply:
`commites` => `commits`